libtenzir/src/index.cpp

```diff
@@ -1167,12 +1089,22 @@ index(index_actor::stateful_pointer<index_state> self,
     return index_actor::behavior_type::make_empty_behavior();
   }
   self->set_exit_handler([self](const caf::exit_msg& msg) {
-    TENZIR_DEBUG("{} received EXIT from {} with reason: {}", *self, msg.source,
-                 msg.reason);
-    for (auto&& [rp, _] : std::exchange(self->state.delayed_queries, {}))
+    TENZIR_VERBOSE("{} received EXIT from {} with reason: {}", *self,
+                   msg.source, msg.reason);
+    for (auto&& [rp, _] : std::exchange(self->state.delayed_queries, {})) {
       rp.deliver(msg.reason);
-    // Flush buffered batches and end stream.
-    self->state.stage->stop();
+    }
+    self->state.shutting_down = true;
+    self
+      ->request(static_cast<index_actor>(self), std::chrono::seconds{120},
+                atom::flush_v)
+      .then(
+        [self]() {
+          self->quit();
+        },
+        [self](caf::error& err) {
+          self->quit(err);
```

Comment:
Please add some context here, and provide a special error message if `err == caf::sec::request_timeout` that is very visible.